pageserver/src/tenant_mgr.rs

```diff
@@ -399,6 +400,26 @@ pub fn list_tenants() -> Vec<TenantInfo> {
         .collect()
 }
 
+/// Check if a given timeline is "broken" \[1\].
+/// The function returns an error if the timeline is "broken".
+///
+/// \[1\]: it's not clear now how should we classify a timeline as broken.
+/// A timeline is categorized as broken when any of following conditions is true:
+/// - failed to load the timeline's metadata
+/// - the timeline's disk consistent LSN is zero
+fn check_broken_timeline(repo: &LayeredRepository, timeline_id: ZTimelineId) -> anyhow::Result<()> {
+    let metadata = load_metadata(repo.conf, timeline_id, repo.tenant_id())
+        .context("failed to load metadata")?;
+
+    // A timeline with zero disk consistent LSN can happen when the page server
+    // failed to checkpoint the timeline import data when creating that timeline.
+    if metadata.disk_consistent_lsn() == Lsn(0) {
```

Comment:
NIT: we have `Lsn::INVALID` that can be uses on the rhs here.

Reply:
I cannot find `Lsn::INVALID`. Which file are you referring to?

Reply:
https://github.com/neondatabase/neon/blob/0e1bd57c533165dbe4bead8fa23baefa09c97b82/libs/utils/src/lsn.rs#L30

Might have slipped past by your PR, in that case a rebase is needed to get it here.